mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -37,113 +45,55 @@ def __init__(self, source, matrix='enterprise', local=None):
-            temp = self.source_handle.query(filts)
-            for b in temp:
-                for c in b['external_references']:
-                    if c['source_name'] == "mitre-attack" and c['external_id'] == match:
-                        return a
-        else:
-            filts = [Filter('type', '=', 'intrusion-set'),
-                     Filter('aliases', '=', match)]
-            a = self.source_handle.query(filts)
-        if len(a):
-            return a[0]
-        else:
-            raise UnableToFindGroup
-
-    def get_software(self, match):
+    def get_stix_object(self, match):
         """
-        Retrieve software for a given match
-        :param match: matching pattern
-        :return: first matching software or UnableToFindSoftware exception
+        Retrieve the stix object for a given string
+        :param match: The string to match on - can be a name, alias, or ATT&CK ID
+        :return: the corresponding stix object
         """
         filts = [
-            [Filter('type', '=', 'malware')],
-            [Filter('type', '=', 'tool')]
+            [Filter('name', '=', match)],
+            [Filter(match, 'in', 'aliases')],
+            [Filter(match, 'in', 'x_mitre_aliases')],
+            [Filter('external_references.external_id', '=', match)],
         ]
-        all_software = list(chain.from_iterable(self.source_handle.query(f) for f in filts))
-        for x in all_software:
-            for y in x['external_references']:
-                if y['source_name'] == 'mitre-attack' and y['external_id'] == match:
-                    return x
-            if match == x['name']:
-                return x
-            if 'aliases' in x:
-                if match in x['aliases']:
-                    return x
-        raise UnableToFindSoftware
+        data = list(chain.from_iterable(self.source_handle.query(f) for f in filts))
+        data = remove_revoked(data)
+        if len(data) == 1:
+            return data[0]
+        raise UnableToFindStixObject
 
-    def get_mitigation(self, match):
+    def get_matrix_data(self, match_pattern):
         """
-        Retrieve mitigation for a given match
-        :param match: matching pattern
-        :return: first matching mitigation or UnableToFindMitigation exception
-        """
-        mitigation_list = self.source_handle.query([Filter('type', '=', 'course-of-action')])
-        for x in mitigation_list:
-            for y in x['external_references']:
-                if y['source_name'] == 'mitre-attack' and y['external_id'] == match:
-                    return x
-            if match in x['name']:
-                return x
-        raise UnableToFindMitigation
-
-    def get_matrix_data(self, match_pattern, obj_type):
-        """
-        Retrieve matching attack-pattern objects for match_pattern
+        Retrieve a list of matching attack-pattern (technique) objects for a match_pattern (group, software, mitigation)
```

Comment:
> for a match_pattern (group, software, mitigation)

This is wrong, I think? Match pattern is not a group, software or mitigation, but rather a query to search for. Or maybe it's just misleadingly written. Maybe revise to:
```
Retrieve a list of attack-pattern (technique) objects that map to a group, software or mitigation. 
:param match_pattern: Name, associated group/software (alias), or ATT&CK ID. Techniques mapped to the object matching this pattern are returned.```

Reply:
It's more misleadingly written than straight wrong, since the patch_pattern must be one of those three types of stix_ids. I'll swap in your documentation, though - it makes more sense.